docs/NNNN-mutable-head.md

```diff
@@ -0,0 +1,287 @@
+
+1. Compare the contents of `[object-root]/extensions/NNNN-mutable-head/root-inventory.json.sha512` and `[object-root]/inventory.json.sha512` to ensure that they are the same and that the root object has not been modified since the creation of the mutable HEAD. If their contents are different, there is a version conflict that is up to the implementation to resolve.
+2. Rewrite the inventory manifest and fixity entries that reference `extensions/NNNN-mutable-head/head` to reference `vN` instead, where `vN` is the mutable HEAD version number.
+3. Move `[object-root]/extensions/NNNN-mutable-head/head` to `[object-root]/vN`.
+4. If the directory cannot be moved because there is already a version directory at that location, then there is a version conflict that is up to the implementation to resolve.
+5. Write the updated inventory and sidecar to the object root.
+6. If this fails, move `[object-root]/vN` back to `[object-root]/extensions/NNNN-mutable-head/head` and abort.
+7. Otherwise, write the updated inventory and sidecar to `[object-root]/vN`.
+8. Remove the extension directory.
+
+The following is what the [mutable HEAD inventory example](#inventory) would look like after it is committed:
+
+```json
+{
+  "digestAlgorithm": "sha512",
+  "head": "v2",
+  "id": "ark:/12345/bcd987",
+  "manifest": {
+    "4d27c8...b53": [ "v2/content/r1/foo/bar.xml" ],
+    "9bb43j...n3a": [ "v2/content/r2/file1.txt" ],
+    "u8b99v...7b2": [ "v2/content/r3/file1.txt" ],
+    "7dcc35...c31": [ "v1/content/foo/bar.xml" ],
+    "cf83e1...a3e": [ "v1/content/empty.txt" ],
+    "ffccf6...62e": [ "v1/content/image.tiff" ]
+  },
+  "type": "https://ocfl.io/1.0/spec/#inventory",
+  "versions": {
+    "v1": {
+      "created": "2018-01-01T01:01:01Z",
+      "message": "Initial import",
+      "state": {
+        "7dcc35...c31": [ "foo/bar.xml" ],
+        "cf83e1...a3e": [ "empty.txt" ],
+        "ffccf6...62e": [ "image.tiff" ]
+      },
+      "user": {
+        "address": "alice@example.com",
+        "name": "Alice"
+      }
+    },
+    "v2": {
+      "created": "2018-02-02T02:02:02Z",
+      "message": "Fix bar.xml, remove image.tiff, add empty2.txt",
+      "state": {
+        "9bb43j...n3a": [ "file2.txt" ],
+        "u8b99v...7b2": [ "file1.txt" ],
+        "4d27c8...b53": [ "foo/bar.xml" ],
+        "cf83e1...a3e": [ "empty.txt", "empty2.txt" ]
+      },
+      "user": {
+        "address": "bob@example.com",
+        "name": "Bob"
+      }
+    }
+  }
+}
+```
+
+### Accessing a Mutable HEAD
+
+An object with a mutable HEAD is accessed in much the same way as an object that doesn’t have a mutable HEAD. The primary difference is that the inventory file at `[object-root]/extensions/NNNN-mutable-head/head/inventory.json` is used instead of the inventory file in the object root. Outside of the OCFL object, there is no discernible difference between the mutable HEAD and any other version.
```

Comment:
I am not exactly sure what is being meant by:
> Outside of the OCFL object, ...

Reply:
Fair enough. I was trying to emphasize that for reading purposes the mutable head version behaves exactly the same as a normal version. I will rework this section a little.

Reply:
I think the point is perhaps that an implementation can present the mutable head as if it were the next immutable version

Reply:
I updated that paragraph